src/Models/Product.php

```diff
@@ -259,105 +257,4 @@ public static function exist($productId): bool
     {
         return self::withoutGlobalScopes()->where('entity_id', $productId)->exists();
     }
-
-    protected function makeAllSearchableUsing(Builder $query)
-    {
-        return $query->selectOnlyIndexable()
-            ->with(['categoryProducts', 'reviewSummary'])
-            ->withEventyGlobalScopes('index.product.scopes');
-    }
-
-    public function shouldBeSearchable(): bool
-    {
-        if (! in_array($this->visibility, config('rapidez.indexer.visibility'))) {
-            return false;
-        }
-
-        $showOutOfStock = (bool) Rapidez::config('cataloginventory/options/show_out_of_stock', 0);
-        if (! $showOutOfStock && ! $this->in_stock) {
-            return false;
-        }
-
-        return true;
-    }
-
-    public function searchableData(array $data, Product $product): array
-    {
-        $data['store'] = config('rapidez.store');
-
-        // TODO: Can we do this without having to cache this?
-        $maxPositions = Cache::driver('array')->rememberForever('max-positions-' . config('rapidez.store'), function () {
-            return CategoryProduct::query()
-                ->selectRaw('GREATEST(MAX(position), 0) as position')
-                ->addSelect('category_id')
-                ->groupBy('category_id')
-                ->pluck('position', 'category_id');
-        });
-
-        foreach ($product->super_attributes ?: [] as $superAttribute) {
-            $data['super_' . $superAttribute->code] = $superAttribute->text_swatch || $superAttribute->visual_swatch
-                ? array_keys((array) $product->{'super_' . $superAttribute->code})
-                : Arr::pluck($product->{'super_' . $superAttribute->code} ?: [], 'label');
-        }
-
-        $data = $this->withCategories($data);
-
-        $data['positions'] = $product->categoryProducts
-            ->pluck('position', 'category_id')
-            // Turn all positions positive
-            ->mapWithKeys(fn ($position, $category_id) => [$category_id => $maxPositions[$category_id] - $position]);
-
-        return $data;
-    }
-
-    // TODO: With this and all methods above this model
-    // is getting pretty big... can we move this?
-    public function withCategories(array $data): array
-    {
-        // TODO: Do we still need to do this in this way? Can we refactor this to be nicer now that we're using Scout?
```

Comment:
We could solve this using a relationship on the product. But it would be less efficient